titus-server-gateway/src/main/java/com/netflix/titus/gateway/eviction/EvictionResource.java

```diff
@@ -0,0 +1,140 @@
+import com.netflix.titus.runtime.connector.eviction.EvictionServiceClient;
+import com.netflix.titus.runtime.endpoint.common.rest.Responses;
+import com.netflix.titus.runtime.eviction.endpoint.grpc.GrpcEvictionModelConverters;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import rx.Observable;
+
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+@Api(tags = "Eviction service")
+@Path("/v3/eviction")
+@Singleton
+public class EvictionResource {
+
+    private final EvictionServiceClient evictionServiceClient;
+
+    @Inject
+    public EvictionResource(EvictionServiceClient evictionServiceClient) {
+        this.evictionServiceClient = evictionServiceClient;
+    }
+
+    @ApiOperation("Return the global disruption budget")
+    @Path("budgets/global")
+    @GET
+    public SystemDisruptionBudget getGlobalDisruptionBudget() {
+        return Responses.fromSingleValueObservable(
+                evictionServiceClient
+                        .getDisruptionBudget(Reference.global())
+                        .map(GrpcEvictionModelConverters::toGrpcSystemDisruptionBudget)
+        );
+    }
+
+    @ApiOperation("Return a tier disruption budget")
+    @Path("budgets/tiers/{tier}")
+    @GET
+    public SystemDisruptionBudget getTierDisruptionBudget(@PathParam("tier") String tier) {
+        return Responses.fromSingleValueObservable(
+                evictionServiceClient
+                        .getDisruptionBudget(Reference.tier(StringExt.parseEnumIgnoreCase(tier, Tier.class)))
+                        .map(GrpcEvictionModelConverters::toGrpcSystemDisruptionBudget)
+        );
+    }
+
+    @ApiOperation("Return a capacity group disruption budget")
+    @Path("budgets/capacityGroups/{name}")
+    @GET
+    public Observable<SystemDisruptionBudget> getCapacityGroupDisruptionBudget(@PathParam("name") String capacityGroupName) {
+        return Responses.fromSingleValueObservable(
+                evictionServiceClient
+                        .getDisruptionBudget(Reference.capacityGroup(capacityGroupName))
+                        .map(GrpcEvictionModelConverters::toGrpcSystemDisruptionBudget)
+        );
+    }
+
+    @ApiOperation("Return the global eviction quota")
+    @Path("quotas/global")
+    @GET
+    public EvictionQuota getGlobalEvictionQuota() {
+        return Responses.fromSingleValueObservable(
+                evictionServiceClient
+                        .getEvictionQuota(Reference.global())
+                        .map(GrpcEvictionModelConverters::toGrpcEvictionQuota)
+        );
+    }
+
+    @ApiOperation("Return a tier eviction quota")
+    @Path("quotas/tiers/{tier}")
+    @GET
+    public EvictionQuota getTierEvictionQuota(@PathParam("tier") String tier) {
+        return Responses.fromSingleValueObservable(
+                evictionServiceClient
+                        .getEvictionQuota(Reference.tier(StringExt.parseEnumIgnoreCase(tier, Tier.class)))
```

Comment:
maybe this is fine but passing in an invalid tier will result in a Server Internal Error here.

Reply:
I still did not define all error codes and exception mappings. I will do this later when the API stabilizes.